core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -150,8 +150,14 @@ void SYCLInternal::initialize(const sycl::queue& q) {
 
     m_maxShmemPerBlock =
         d.template get_info<sycl::info::device::local_mem_size>();
-    m_indirectKernelMem.reset(*m_queue, m_instance_id);
+
     m_indirectReducerMem.reset(*m_queue, m_instance_id);
+    for (auto& usm_mem : m_indirectKernelMem) {
+      usm_mem.reset(*m_queue, m_instance_id);
+      // TODO 0x1440= 5184, arbitrary, larger than largest encountered kernel.
+      usm_mem.reserve(0x1440);
```

Comment:
I'm not quite sure that we preallocate memory. Do you have a good reason for that?

Reply:
As discussed offline, I think we need to see some performance figures for typical cases, in case repeated reallocation hurts performance.